include/circt/Dialect/SV/SVInOutOps.td

```diff
@@ -176,3 +176,30 @@ def IndexedPartSelectInOutOp : SVOp<"indexed_part_select_inout",
   let assemblyFormat = "$input`[`$base (`decrement` $decrement^)?`:` $width`]`"
                         " attr-dict `:` type($input) `,` type($base)";
 }
+
+def InOutStructType
+  : Type<CPred<"getInOutElementType($_self).isa<hw::StructType>()">,
+         "an inout type with struct field", "::circt::hw::InOutType">;
+
+def StructFieldInOutOp : SVOp<"struct_field_inout",
+                              [NoSideEffect, InferTypeOpInterface]> {
+  let summary = "Create an subfield inout memory to produce an inout element.";
+  let description = "See SV Spec 7.2.";
+
+  let arguments = (ins InOutStructType:$input, StrAttr:$field);
```

Comment:
We had been converting to using integer indexes into structs to reduce verification.

Reply:
but I guess that's only in FIRRTL still, so nevermind.